### ISSUE TITLE
RestAPI endpoint "send-verify-email" sending execute actions email template.

### ISSUE THREAD

**Issue by osama-ahmed-wiz** [keycloak/keycloak#15190]
### Area

admin/api

### Describe the bug

The issue is when I use the send-verify-endpoint for email verification email, It triggers the email for execute actions. 
I'm using keycloak admin client package "@keycloak/keycloak-admin-client"

Endpoint:
![image](https://user-images.githubusercontent.com/106302788/198338361-6a4c2aa1-f257-4465-ad54-875df6094a92.png)

Email
![image](https://user-images.githubusercontent.com/106302788/198339453-1ec7bff4-c6ed-466d-b0e7-e76098d103c9.png)

### Version

18.0.2

### Expected behavior

It should send email verification template which will verify user email and redirect.

### Actual behavior

It was sending execute actions email template which redirecting user to perform actions.

### How to Reproduce?

API Endpoint: PUT /{realm}/users/{id}/send-verify-email

**Comment by lexcao** [comment]
Hi, @osama-ahmed-wiz 
Currently, the `/send-verify-email` endpoint is just to send `VERIFY_EMAIL` execution action email then executing verifing.
It is same to the admin UI operation like below.
<img width="720" alt="image" src="https://user-images.githubusercontent.com/14901139/200115272-28f94ef5-e6ff-4058-aa2b-c4e8e0b96708.png">

---
If you want to a verify email instead of action email, you could try this required aciton.
```
PUT /admin/realms/{realm}/users/{user_id} HTTP/1.1
{ 	
    "requiredActions": ["VERIFY_EMAIL"]
}
```
It is same to the admin UI operation like below.
<img width="720" alt="image" src="https://user-images.githubusercontent.com/14901139/200115310-2b15b166-e602-4ca9-9be9-e1b4facc73c0.png">

**Comment by osama-ahmed-wiz** [comment]
Yes, I want to a verify email instead of action email.

If I use the following endpoint
PUT /admin/realms/{realm}/users/{user_id} HTTP/1.1
{ 	
    "requiredActions": ["VERIFY_EMAIL"]
}
it just sets the required user actions. Its working fine.



Now I want to execute the set action through API. Is there any end point to execute the set actions?

PUT /{realm}/users/{id}/execute-actions-email
{
    actions: [ VERIFY_EMAIL ] 
}
Above endpoint also send the execute action email.

I'm using openid-connect endpoint for login and token and not using client side login theme/ui. So is it possible to execute only set actions email.?

**Comment by lexcao** [comment]
>  So is it possible to execute only set actions email.?

AFAK, nope.

**Comment by osama-ahmed-wiz** [comment]
So basically we can not send **email-verification** email directly from REST API?

**Comment by mposolda** [comment]
Thanks for the report, but unfortunately due the amount of other reported issues and other priorities, Keycloak team does not have time to properly triage this bug. So preliminary added to Backlog for now.
It will be helpful if:
- You can verify if still applicable in latest Keycloak released version. If not, then it is welcome to close this issue.
- If you figure that this may not be a valid bug (for example just a mistake in configuration etc), it will be also welcome to close this issue
- If still applicable in latest version, it will be welcome to add the comment as well, that this was still reproduced with latest Keycloak version as it is very valuable info. Anyone is welcome to comment with this or add other relevant comments to this issue.

**Comment by lauritzh** [comment]
Jumping in as I was looking for a possibility to send localized emails and stumbled over this issue.

> So basically we can not trigger/send email-verification email directly from REST API?

For me, the following REST API call can be used to trigger "verify email address" emails:
```http
PUT /admin/realms/[REALM]/users/[ID]/execute-actions-email HTTP/1.1
Host: example.com
Content-Length: 16
Content-Type: application/json
Authorization: bearer eyJh[...]

["VERIFY_EMAIL"]
```

Looks like the previously mentioned POST body within this issue did not match the currently expected format.

I would highly appreciate a heads-up in case anybody knows a solution to specify which locale the resulting email should use 😄 

Best regards and have a great day,
Lauritz

**Comment by ronny-standtke** [comment]
@lauritzh I was not able to send an email for verification. I also receive an "execute actions" email.
@mposolda I'm running latest version (21.0.2) in a docker container in dev mode.

Following procedure

1. adding a user via "POST /{realm}/users"
2. add roles to user "POST /{realm}/users/{id}/role-mappings/realm"
3. assign to group "PUT /{realm}/users/{id}/groups/{groupId}"
4. finally "PUT /{realm}/users/${id}/send-verify-email"

I expect that I receive an email to verify my email like I would receive when using registration process.

**Comment by JapuDCret** [comment]
Linking https://issues.redhat.com/browse/KEYCLOAK-18455 as this has already been raised in 2021.

We have the same problem using Keycloak `send-verify-email18.0.2`:
 * `send-verify-email` adds the required action `VERIFY_EMAIL` and triggers a generic "please execute required actions" e-mail instead of the specific "please verify e-mail" email

**Comment by DaBonse** [comment]
We have the same problem.

We have upgraded from 15.0.2 to 21+. In 15.0.2 we used this workflow:

1. Create user over RestApi
2. Trigger verification email over RestApi in this way:

PUT /auth/admin/realms/<realmName>/users/<userId>/send-verify-email

with payload:

`['VERIFY_EMAIL']`

3. User got an email
4. user click on link in email
5. user email was verified and user was logged in

Since version 21 (we have jumped from 15 to 21) this does not work anymore. Instead, the email was send, but the verification link does nothing. After clicking, user has to log in and than the verification email comes again and only this link verified the email.

What we need is: Send a email, to verify email address, over RestAPI.
